metricbeat/module/ceph/osd_tree/osd_tree_integration_test.go

```diff
@@ -22,13 +22,20 @@ import (
 	"os"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
 func TestData(t *testing.T) {
-	f := mbtest.NewEventsFetcher(t, getConfig())
-	err := mbtest.WriteEvents(f, t)
-	if err != nil {
+	f := mbtest.NewReportingMetricSetV2(t, getConfig())
+	events, errs := mbtest.ReportingFetchV2(f)
```

Comment:
Again, this is the only critical change needed here. Check my comments [here](https://github.com/elastic/beats/pull/11049#discussion_r263344371) to avoid a flaky test